django/db/migrations/state.py

```diff
@@ -548,7 +549,7 @@ def render(self, apps):
         body.update(self.construct_managers())
 
         # Then, make a Model object (apps.register_model is called in __new__)
-        return type(
+        return metaclassmaker()(
```

Comment:
Did you check if it is causing performance problem? As for great part of models it just fallback to `type()`. If it's really such a problem I can profile it and try speed up, just give me info how much.